src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModuleResolutionFunction.java

```diff
@@ -209,38 +267,12 @@ private static RepoSpec maybeAppendAdditionalPatches(RepoSpec repoSpec, ModuleOv
         .build();
   }
 
-  @Nullable
-  private static RepoSpec computeRepoSpec(
-      InterimModule interimModule, ModuleOverride override, ExtendedEventHandler eventHandler)
-      throws BazelModuleResolutionFunctionException, InterruptedException {
-    if (interimModule.getRegistry() == null) {
-      // This module has a non-registry override. We don't need to store the repo spec in this case.
-      return null;
-    }
-    try {
-      RepoSpec moduleRepoSpec =
-          interimModule
-              .getRegistry()
-              .getRepoSpec(
-                  interimModule.getKey(), interimModule.getCanonicalRepoName(), eventHandler);
-      return maybeAppendAdditionalPatches(moduleRepoSpec, override);
-    } catch (IOException e) {
-      throw new BazelModuleResolutionFunctionException(
-          ExternalDepsException.withMessage(
-              Code.ERROR_ACCESSING_REGISTRY,
-              "Unable to get module repo spec from registry: %s",
-              e.getMessage()),
-          Transience.PERSISTENT);
-    }
-  }
-
   /**
    * Builds a {@link Module} from an {@link InterimModule}, discarding unnecessary fields and adding
    * extra necessary ones (such as the repo spec).
    */
   static Module moduleFromInterimModule(
-      InterimModule interim, ModuleOverride override, ExtendedEventHandler eventHandler)
-      throws BazelModuleResolutionFunctionException, InterruptedException {
+      InterimModule interim, ModuleOverride override, RepoSpec remoteRepoSpec) {
```

Comment:
please add a small note about when `remoteRepoSpec` is not null (and ideally what "remote" means here).